packages/core-utils/src/watcher.ts

```diff
@@ -73,22 +73,31 @@ export class Watcher {
     msgHash: string,
     pollForPending: boolean = true
   ): Promise<TransactionReceipt> {
-    const blockNumber = await layer.provider.getBlockNumber()
-    const startingBlock = Math.max(blockNumber - this.NUM_BLOCKS_TO_FETCH, 0)
-    const successFilter = {
+    let matches: ethers.providers.Log[] = []
+    const successFilter: ethers.providers.Filter = {
       address: layer.messengerAddress,
       topics: [ethers.utils.id(`RelayedMessage(bytes32)`)],
-      fromBlock: startingBlock,
     }
-    const failureFilter = {
+    const failureFilter: ethers.providers.Filter = {
```

Comment:
Since you got rid of the `provider.on` event listeners below, you could move `successFilter` and `failureFilter` into the `while` loop that you constructed. Would get rid of a little code.